### PR TITLE
Add RISCV vector extension

### PR DESCRIPTION
**Pull request by michael-roe**

### CODE REVIEW

File: include/cpuinfo_riscv.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ typedef struct {
   int D : 1;         // Standard Extension for Double-Precision Floating-Point
   int Q : 1;         // Standard Extension for Quad-Precision Floating-Point
   int C : 1;         // Standard Extension for Compressed Instructions
+  int V : 1;         // Standard Extneions for Vector Instructions
```
Comment (Mizux) [inline include/cpuinfo_riscv.h:39]: Extension

**Comment by michael-roe** [comment]
Ok, I've fixed the typo in the comment.

**Comment by gchatelet** [comment]
@michael-roe I originally implemented only the ratified extensions from [the spec](https://riscv.org/technical/specifications/), that's why vector extension is missing.

Do you know of any board that currently supports vector extensions?
Also can you add a test case for it?

**Comment by michael-roe** [comment]
Yes, I can add a test case. (I was kind of expecting you to ask for one...)

The QEMU emulator can emulate V, and there are implementations on FPGA.

As far as I know, current boards don't support it (Allwinner D1 is based on an obsolete version of the vector spec), but things are moving  fairly quickly.

**Comment by Mizux** [comment]
Last kernel news about it AFAIK
https://lore.kernel.org/lkml/802edef897b4bd40f3ebc2cb47c02b8e2f61a666.1636362169.git.greentime.hu@sifive.com/

**Comment by michael-roe** [comment]
Ok, I've added tests for the V extension.

File: test/cpuinfo_riscv_test.cc [hunk 2]
```diff
@@ -150,6 +153,27 @@ uarch     : sifive,bullet0)");
   EXPECT_TRUE(info.features.D);
   EXPECT_FALSE(info.features.Q);
   EXPECT_TRUE(info.features.C);
+  EXPECT_FALSE(info.features.V);
+}
+
+TEST(CpuinfoRiscvTest, QemuCpuInfo) {
+	  ResetHwcaps();
+	    auto& fs = GetEmptyFilesystem();
+	      fs.CreateFile("/proc/cpuinfo", R"(
```
Comment (toor1245) [inline test/cpuinfo_riscv_test.cc:162]: could you format these lines?

Comment (michael-roe) [reply test/cpuinfo_riscv_test.cc]: Ok, I've fixed the indentation.

**Comment by michael-roe** [comment]
Is this PR Ok now?

(There are a bunch of people who want rto use riscv vector)

**Comment by gchatelet** [comment]
Sorry for the lag and thx for the PR!

**Comment by Mizux** [comment]
please notice https://elixir.bootlin.com/linux/latest/source/arch/riscv/include/asm/hwcap.h still **does not** contain it so we could change this implementation if the kernel want to go in a different way...

**Pull request merged**